File: c/gpt/p10.c
// Task p10: getValueFromArray-like behavior: read index from stdin and exit with the value
#include <stdio.h>
#include <stdlib.h>

int getValueFromArray(int *arr, int n, int idx)
{
    if (idx < 0 || idx >= n)
        return -1;
    return arr[idx];
}

int main(void)
{
    int sequence[] = {10, 20, 30, 40, 50};
    int n = sizeof(sequence) / sizeof(sequence[0]);
    int idx;
    if (scanf("%d", &idx) != 1)
    {
        fprintf(stderr, "Failed to read index from stdin\n");
        return 1;
    }
    int val = getValueFromArray(sequence, n, idx);
    if (val < 0)
    {
        printf("-1\n");
        return 255; // map -1 to exit 255
    }
    printf("%d\n", val);
    return val; // may not be a valid exit code if large but OK for this example
}

File: javascript/gpt/p10.js
// Task p10: getValueFromArray and return value as process exit code
function getValueFromArray(arr, size, index) {
  if (index < 0 || index >= size) return -1;
  return arr[index];
}

// Read integer from stdin and call getValueFromArray
const readline = require("readline");

const id_sequence = [10, 20, 30, 40, 50];

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});
let input = "";
rl.on("line", (line) => {
  input += line;
});
rl.on("close", () => {
  const s = input.trim();
  const idx = parseInt(s, 10);
  const result = getValueFromArray(
    id_sequence,
    id_sequence.length,
    Number.isNaN(idx) ? -1 : idx,
  );
  // Node exit codes are 0-255; normalize negative to 255
  const exitCode = result < 0 ? 255 : result & 0xff;
  process.exit(exitCode);
});
